### PR TITLE
160 add stress and strain state calculation

### PR DESCRIPTION
**Pull request by axtimhaus**

### CODE REVIEW

**Comment by axtimhaus** [comment]
Current draft adds stresses and strains in x, y and z direction. Would it be more feasible to use terms like longitudinal, vertical and lateral? Think of 3-roll and 4-roll passes where the cartesian coordinates have no useful meaning.

**Comment by ChRen95** [comment]
I think the usage of longitudinal, lateral and vertical is better. We should of course make feasible drawings for the docs to explain these concepts.

**Comment by ChRen95** [comment]
From my side, this PR is ready.
Hookimpls should be provided via plugins.

**Comment by axtimhaus** [comment]
> Current draft adds stresses and strains in x, y and z direction. Would it be more feasible to use terms like longitudinal, vertical and lateral? Think of 3-roll and 4-roll passes where the cartesian coordinates have no useful meaning.

Question still open

**Comment by ChRen95** [comment]
ready for merging, @axtimhaus please check.

File: pyroll/core/profile/hookimpls.py [hunk 1]
```diff
@@ -68,3 +68,19 @@ def heat_penetration_number(self: Profile):
 def thermal_diffusivity(self: Profile):
     if hasattr(self, "thermal_conductivity") and hasattr(self, "density") and hasattr(self, "specific_heat_capacity"):
         return self.thermal_conductivity / (self.density * self.specific_heat_capacity)
+
+
+@Profile.hydrodynamic_stress
+def hydrodynamic_stress(self: Profile):
```
Comment (axtimhaus) [inline pyroll/core/profile/hookimpls.py:74]:
```suggestion
@Profile.hydrostatic_stress
def hydrostatic_stress(self: Profile):
```

File: pyroll/core/profile/profile.py [hunk 2]
```diff
@@ -106,6 +106,30 @@ class Profile(HookHost):
...
+    longitudinal_strain = Hook[float]()
+    """Logarithmic normal strain (principal strain) in rolling (x) direction."""
+
+    altitudinal_strain = Hook[float]()
+    """Logarithmic normal strain (principal strain) in thickness (y) direction."""
+
+    latitudinal_strain = Hook[float]()
+    """Logarithmic normal strain (principal strain) in width (z) direction."""
+
+    hydrodynamic_stress = Hook[float]()
```
Comment (axtimhaus) [inline pyroll/core/profile/profile.py:127]:
```suggestion
    hydrostatic_stress = Hook[float]()
```

**Review by ChRen95 — APPROVED** [review]

**Pull request merged**